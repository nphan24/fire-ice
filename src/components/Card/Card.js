import React, { Component } from 'react';
import { fetchMembers } from '../../ApiCalls/fetchMembers';
import { connect } from 'react-redux';
import * as actions from '../../actions';

export const Card = (props) => {
  console.log('prop cards', props);
  const { name, founded, seats, titles, coatOfArms, region, ancestralWeapons, swornMembers } = props.house;

  const seat = seats.map(seat => {
    if (seat === '') {
      return 'NA';
    } else {
      return seat;
    }
  });

  const title = titles.map(title => {
    if (title === '') {
      return 'NA';
    } else {
      return title;
    }
  });

  const ancestralWeapon = ancestralWeapons.map(weapon => {
    if (title === '') {
      return 'NA';
    } else {
      return weapon;
    }
  });

  const handleClick = async (swornMembers) => {
    const members = await fetchMembers(swornMembers);
    console.log('members', members); 
    if (members.length > 1) {
      this.props.addMembers(members);
    }
  };

  return <div>
    <h2>{name}</h2>
    <p>Founded: {founded}</p>
    <p>seats: {seat}</p>
    <p>titles: {title}</p>
    <p>coatOfArms: {coatOfArms}</p>
    <p>region: {region}</p>
    <p>Ancestral Weapon: {ancestralWeapon}</p>
    <button onClick={() => handleClick({swornMembers})} >See Sworn Members</button>
  </div>;
};

const mapStateToProps = state => ({
  members: state.members
});

const mapDispatchToProps = dispatch => ({
  addMembers: (members) => dispatch(actions.addMembers(members))
});

export default connect(mapStateToProps, mapDispatchToProps)(Card);