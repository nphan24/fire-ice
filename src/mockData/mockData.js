export const Houses = [
  {
    name: 'Stark'
  },
  {
    name: 'Lannister'
  }
];

export const Members = [
  {
    name: 'Stark'
  },
  {
    name: 'Lannister'
  }
];

export const data = [
  {
    ancestralWeapons: [''],
    cadetBranches: [],
    coatOfArms:
      'A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)',
    currentLord: '',
    diedOut: '',
    founded: '',
    founder: '',
    heir: '',
    name: 'House Algood',
    overlord: 'https://www.anapioficeandfire.com/api/houses/229',
    region: 'The Westerlands',
    seats: [''],
    swornMembers: [],
    titles: [''],
    url: 'https://www.anapioficeandfire.com/api/houses/1',
    words: ''
  },
  {
    ancestralWeapons: [''],
    cadetBranches: [],
    coatOfArms: 'Gyronny Gules and Sable, a hand couped Or',
    currentLord: 'https://www.anapioficeandfire.com/api/characters/298',
    diedOut: '',
    founded: '',
    founder: '',
    heir: 'https://www.anapioficeandfire.com/api/characters/1922',
    name: 'House Allyrion of Godsgrace',
    overlord: 'https://www.anapioficeandfire.com/api/houses/285',
    region: 'Dorne',
    seats: ['Godsgrace'],
    swornMembers: (4)[
      ('https://www.anapioficeandfire.com/api/characters/298',
      'https://www.anapioficeandfire.com/api/characters/1129',
      'https://www.anapioficeandfire.com/api/characters/1301',
      'https://www.anapioficeandfire.com/api/characters/1922')
    ],
    titles: [''],
    url: 'https://www.anapioficeandfire.com/api/houses/2',
    words: 'No Foe May Pass'
  }
];

export const house = {
  ancestralWeapons: [''],
  coatOfArms:
    'A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)',
  founded: '',
  founder: '',
  name: 'House Algood',
  region: 'The Westerlands',
  seats: [''],
  swornMembers: [],
  titles: [''],
  url: 'https://www.anapioficeandfire.com/api/houses/1',
  words: ''
};
