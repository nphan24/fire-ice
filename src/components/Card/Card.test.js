import { Card } from './Card';
import React from 'react';
import { shallow } from 'enzyme';
import * as mock from '../../mockData/mockData';

describe('Card', () => {
  let wrapper;
  let house = mock.house

  beforeEach(() => {
    wrapper = shallow(<Card house={house}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});