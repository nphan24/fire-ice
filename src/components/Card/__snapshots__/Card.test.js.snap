// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    house={
        Object {
            "ancestralWeapons": Array [
              "",
            ],
            "coatOfArms": "A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)",
            "founded": "",
            "founder": "",
            "name": "House Algood",
            "region": "The Westerlands",
            "seats": Array [
              "",
            ],
            "swornMembers": Array [],
            "titles": Array [
              "",
            ],
            "url": "https://www.anapioficeandfire.com/api/houses/1",
            "words": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          House Algood
</h2>,
        <p>
          Founded: 
          
</p>,
        <p>
          seats: 
          NA
</p>,
        <p>
          titles: 
          NA
</p>,
        <p>
          coatOfArms: 
          A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)
</p>,
        <p>
          region: 
          The Westerlands
</p>,
        <p>
          Ancestral Weapon: 
          
</p>,
        <button
          onClick={[Function]}
>
          See Sworn Members
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "House Algood",
        },
        "ref": null,
        "rendered": "House Algood",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Founded: ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Founded: ",
          "",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "seats: ",
            Array [
              "NA",
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "seats: ",
          "NA",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "titles: ",
            Array [
              "NA",
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "titles: ",
          "NA",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "coatOfArms: ",
            "A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)",
          ],
        },
        "ref": null,
        "rendered": Array [
          "coatOfArms: ",
          "A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "region: ",
            "The Westerlands",
          ],
        },
        "ref": null,
        "rendered": Array [
          "region: ",
          "The Westerlands",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Ancestral Weapon: ",
            Array [
              "",
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "Ancestral Weapon: ",
          "",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "See Sworn Members",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "See Sworn Members",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            House Algood
</h2>,
          <p>
            Founded: 
            
</p>,
          <p>
            seats: 
            NA
</p>,
          <p>
            titles: 
            NA
</p>,
          <p>
            coatOfArms: 
            A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)
</p>,
          <p>
            region: 
            The Westerlands
</p>,
          <p>
            Ancestral Weapon: 
            
</p>,
          <button
            onClick={[Function]}
>
            See Sworn Members
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "House Algood",
          },
          "ref": null,
          "rendered": "House Algood",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            "",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "seats: ",
              Array [
                "NA",
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "seats: ",
            "NA",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "titles: ",
              Array [
                "NA",
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "titles: ",
            "NA",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "coatOfArms: ",
              "A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)",
            ],
          },
          "ref": null,
          "rendered": Array [
            "coatOfArms: ",
            "A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "region: ",
              "The Westerlands",
            ],
          },
          "ref": null,
          "rendered": Array [
            "region: ",
            "The Westerlands",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ancestral Weapon: ",
              Array [
                "",
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "Ancestral Weapon: ",
            "",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "See Sworn Members",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "See Sworn Members",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
